js/type_directives.ts

```diff
@@ -72,16 +69,23 @@ export function parseTypeDirectives(
   // work from the last directive backwards for the next `import`/`export`
   // statement
   directives.reverse();
-  const directiveRecords: Record<string, string> = {};
-  for (const { path, start, end } of directives) {
+  const results = new Map<FileReference, string>();
+  for (const { end, fileName, pos } of directives) {
     const searchString = sourceCode.substring(end);
     const maybeMatch = importExportRegEx.exec(searchString);
     if (maybeMatch) {
-      const [, , fromPath] = maybeMatch;
-      directiveRecords[fromPath] = path;
+      const [matchString, , targetFileName] = maybeMatch;
```

Comment:
I'm surprised this is valid JS... but I guess if it passes the linters it's fine.

Reply:
Yup... you can always "skip" elements when destructuring.